cache.go

```diff
@@ -0,0 +1,78 @@
+package lifecycle
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type cache struct {
+	layersDir   string
+	buildpackID string
+	layers      map[string]*LayerMetadata
+}
+
+func readCache(layersDir, buildpackID string) (cache, error) {
+	layers := make(map[string]*LayerMetadata)
+	tomls, err := filepath.Glob(filepath.Join(layersDir, buildpackID, "*.toml"))
+	if err != nil {
+		return cache{}, err
+	}
+	for _, toml := range tomls {
+		name := strings.TrimRight(filepath.Base(toml), ".toml")
+		layers[name], _ = readTOML(toml)
+		if sha, err := ioutil.ReadFile(filepath.Join(layersDir, buildpackID, name+".sha")); err == nil {
+			layers[name].SHA = string(sha)
+		}
+	}
+	return cache{
+		layersDir:   layersDir,
+		buildpackID: buildpackID,
+		layers:      layers,
+	}, nil
+}
+
+type cacheType int
+
+const (
+	staleNoMetadata cacheType = iota
```

Comment:
I'd suggest prefixing these. Not sure we want a global constant called `valid`. Maybe:
```
layerValid = iota
layerStaleNoMetadata
layerStaleWrongSHA
layerNotForLaunch
```

Reply:
I like the suggestion but how do you feel about prefixing with `cache` instead?

Reply:
Definitely like `cacheX` better!